### PR TITLE
Docs: Add CodeQL system requirements page

### PR DESCRIPTION
**Pull request by adityasharad**
Add a table of platform requirements (OS, OS version, CPU architecture).
Add a list of additional language-specific software requirements.

Preview:
Screen Shot 2022-09-28 at 11 58 24 AM

### CODE REVIEW

**Review by tausbn — DISMISSED** [review]
Python 👍

File: docs/codeql/codeql-overview/system-requirements.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Additional software requirements
+################################
+
+- For extraction of compiled languages on all platforms, the system must be configured so that it can successfully build and compile your code, independently of CodeQL.
+- On Linux, extraction of compiled languages (C/C++, C#, Go, Java) and Ruby requires ``glibc`` version 2.17 or greater.
+    - Extraction of these languages on ``musl-c``-based Linux distributions, such as Alpine Linux, is not supported.
+- TypeScript extraction on all platforms requires Node.js to be installed and available on the ``PATH`` as ``node``.
+- On Linux and macOS, extraction of Python 2 or Python 3 requires Python 3 to be installed and available on the ``PATH`` as ``python3`` or ``python``.
+        - For Python 2 extraction, we also recommend having Python 2 installed and available on the ``PATH`` as ``python2``.
+- On Windows, extraction of Python 2 or Python 3 requires the Python launcher to be installed and available on the ``PATH`` as ``py.exe``.
```
Comment (felicitymay) [inline docs/codeql/codeql-overview/system-requirements.rst:24]: Suggest we might reformat something like this:

```suggestion
To generate a CodeQL database for a compiled language, you must ensure that the system can successfully build and compile your code, independently of CodeQL. 

CodeQL tools have the following requirements.

For compiled languages (C/C++, C#, Go, Java) and Ruby on Linux:

- ``glibc`` version 2.17 or greater must be installed.
- ``musl-c``-based Linux distributions, such as Alpine Linux, are not supported.

For Typescript on all platforms:

- Node.js must be installed and available on the ``PATH`` as ``node``.

For Python:
- On Linux and macOS, Python 3 must be installed and available on the ``PATH`` as ``python3`` or ``python``.
- For Python 2 extraction on Linux and macOS, we also recommend having Python 2 installed and available on the ``PATH`` as ``python2``.
- On Windows, the Python launcher must be installed and available on the ``PATH`` as ``py.exe``.
```

**Review by felicitymay — APPROVED** [review]
@adityasharad - thank you so much for rounding up this missing information 💖 

I've made a couple of formatting suggestions, but generally this looks good 👍🏻

File: docs/codeql/support/reusables/platforms.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+   macOS 12 Monterey","x86-64
+
+   x86-64, arm64 (Apple Silicon) [1]_
+
+   x86-64, arm64 (Apple Silicon) [1]_"
+
+.. container:: footnote-group
+
+    .. [1] Support for Apple Silicon is currently in beta.
```
[the remark this answers is not in the document]
Comment (adityasharad) [reply docs/codeql/support/reusables/platforms.rst:29]: Or just put it on 25, so it appears below both mentions of Apple Silicon?

**Comment by adityasharad** [comment]
Thanks! I have addressed all comments (with minor tweaks). This will need a fresh approval, sorry.

**Review by tausbn — APPROVED** [review]
Python still 👍

**Pull request merged**